src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/ServiceController.php

```diff
@@ -143,7 +148,40 @@ public function create(Request $request): RedirectResponse|Response
             $this->logger->info(sprintf('Save new Service, service was created by: %s', '@todo'), (array) $command);
             try {
                 $this->commandBus->handle($command);
-                $this->addFlash('info', 'service.create.flash.success');
+
+                try {
+                    $service = $this->serviceService->getByLastInsertId();
+
+                    if ($service === null) {
+                        throw new InviteException('Could not find created Service.');
+                    }
+
+                    $message = $translator->trans('service.create.invitationMessage');
+
+                    if ($command->getTeamManagerEmail() === null || $service->getInviteRoleId() === null) {
+                        throw new InviteException('Could not create invite.');
+                    }
```

Comment:
Interesting. 

I do not think the Exception can be thrown, as this is just code to make phpstan happy (rightfully so).
The getTeamManagerEmail cannot return null because the command is validated at this point, but the typing system does not know that.

For newly created services, it cannot happen that the invite role id is empty, as enforced by `\Surfnet\ServiceProviderDashboard\Application\CommandHandler\Service\CreateServiceCommandHandler::handle`.

When this feature goes live, a migration will be made that will populate all services with invite role id's (but this code is only ran on new Services only anyway).

---

The reason I added this in the controller, is that the sending of an invite should in theory not follow the creation of the role. If I add it to the commandhandler, they become inseparable. It would not be possible to use this command for a hypothetical 'resend invite' feature.

It would also intertwine the error handling logic. So I think this solution is preferred.

Reply:
discussed